### PR TITLE
Updates from Tonight

### PR DESCRIPTION
**Pull request by stephencelis**
- Pullback
- Recursive description

### CODE REVIEW

File: Sources/SnapshotTesting/Attachment.swift [hunk 1]
```diff
@@ -5,9 +5,13 @@ public struct Attachment {
   internal let rawValue: XCTAttachment
   #endif
 
-  init(string: String, name: String? = nil) {
+  init(string: String, name: String? = nil, uniformTypeIdentifier: String? = nil) {
```
Comment (stephencelis) [inline Sources/SnapshotTesting/Attachment.swift]: This lets us tag diff artifacts with types...though turns out that QuickLook doesn't have a nice rendering for diffs without a plugin :(

File: Sources/SnapshotTesting/SnapshotTestCase.swift [hunk 2]
```diff
@@ -17,8 +17,8 @@ open class SnapshotTestCase: XCTestCase {
     where A.A == A
   {
     return assertSnapshot(
+      of: A.defaultStrategy,
```
Comment (stephencelis) [inline Sources/SnapshotTesting/SnapshotTestCase.swift]: Changing up the signature per our chat.

File: Sources/SnapshotTesting/Strategy/Any.swift [hunk 3]
```diff
@@ -1,32 +1,8 @@
 import Foundation
 
-extension SnapshotTestCase {
-  public func assertSnapshot(
-    matchingAny value: Any,
```
Comment (stephencelis) [inline Sources/SnapshotTesting/Strategy/Any.swift]: Preferring `assertSnapshot(of: .any, matching)` now.

File: Sources/SnapshotTesting/Strategy/Any.swift [hunk 4]
```diff
@@ -98,3 +73,7 @@ private let snapshotDateFormatter: DateFormatter = {
   formatter.timeZone = TimeZone(abbreviation: "UTC")
   return formatter
 }()
+
+func purgePointers(_ string: String) -> String {
```
Comment (stephencelis) [inline Sources/SnapshotTesting/Strategy/Any.swift]: Can rename, lol. Internal for now.

File: Tests/SnapshotTestingTests/__Snapshots__/SnapshotTestingTests/testNSView.2.txt [hunk 5]
```diff
@@ -0,0 +1,4 @@
+[   AF      LU ] h=--- v=--- NSButton "Push Me" f=(0,0,77,32) b=(-)
```
Comment (stephencelis) [inline Tests/SnapshotTestingTests/__Snapshots__/SnapshotTestingTests/testNSView.2.txt]: NSView's version of `recursiveDescription` is `_subtreeDescription`, which misses the main view. Prob wanna:

- [ ] Embed the view in a superview before calling this method on the superview.

File: Sources/SnapshotTesting/Strategy.swift [hunk 6]
```diff
@@ -41,8 +41,8 @@ public struct Strategy<A, B> {
     }
   }
 
-  public func contramap<A0>(_ transform: @escaping (A0) -> A) -> Strategy<A0, B> {
-    return self.asyncContramap { Async(value: transform($0)) }
+  public func pullback<A0>(_ transform: @escaping (A0) -> A) -> Strategy<A0, B> {
```
Comment (mbrandonw) [inline Sources/SnapshotTesting/Strategy.swift]: 🔙

**Pull request merged**